### PR TITLE
drivers/pinmux: stm32: Use pull-up for _SPI_NSS pins

### LINKED ISSUE #17998: STM32 (Nucleo L476RG) SPI pins floating
Not sure whether that's a bug or simply me holding it wrong, but since there's so little documentation available on this I cannot figure out what's going on. I'm using the following code to test:

```cpp
static const struct spi_config spi_cfg = {
    .frequency = 4000000,
    .operation = SPI_OP_MODE_MASTER | SPI_WORD_SET(8) | SPI_TRANSFER_MSB | SPI_MODE_CPOL,
    // .slave = 0,
};

struct device * spi_dev;

static bool spi_init(void)
{
    spi_dev = device_get_binding(DT_SPI_1_NAME);

    if (spi_dev == NULL) {
        printk("Could not get %s device\n", DT_SPI_1_NAME);
        k_sleep(2000);
        return false;
    }

    return true;
}

uint8_t spi_test_send(uint8_t cmd)
{
    int err;
    static u8_t tx_buffer[1];
    static u8_t rx_buffer[1];

    tx_buffer[0] = cmd;

    const struct spi_buf tx_buf = {
        .buf = tx_buffer,
        .len = sizeof(tx_buffer)
    };
    const struct spi_buf_set tx = {
        .buffers = &tx_buf,
        .count = 1
    };

    struct spi_buf rx_buf = {
        .buf = rx_buffer,
        .len = sizeof(rx_buffer),
    };

    const struct spi_buf_set rx = {
        .buffers = &rx_buf,
        .count = 1
    };

    err = spi_transceive(spi_dev, &spi_cfg, &tx, &rx);
    if (err) {
        printk("SPI error: %d\n", err);
    } else {
        printk("TX sent: %x\n", tx_buffer[0]);
        printk("RX recv: %x\n", rx_buffer[0]);
        return rx_buffer[0];
        // tx_buffer[0]++;
    }
}

uint8_t myData[] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};

int main(void)
{
    bool spiInitialized = false;

    while(!spiInitialized) {
        spiInitialized = spi_init();
    }

    spi_test_send(1);

    for(int i=0; i<10; i++) {
        k_sleep(15);
        myData[i] = spi_test_send(0);
    }

     k_sleep(10000);
}
```

My prj.conf looks like this:

```
CONFIG_SERIAL=y

CONFIG_CPLUSPLUS=y
CONFIG_PRINTK=y
CONFIG_STDOUT_CONSOLE=n
CONFIG_I2C=y
CONFIG_SENSOR=y
CONFIG_MULTITHREADING=y
CONFIG_GPIO=y

CONFIG_PINMUX=y
CONFIG_SPI=y
CONFIG_SPI_1=y
```

After deploying this on the L476RG, I'm measuring the voltage on CN8/A2 (-> PA_4), as this is documented to be [the default CS pin used by Zephyr](https://docs.zephyrproject.org/latest/boards/arm/nucleo_l476rg/doc/index.html#default-zephyr-peripheral-mapping). The pin is floating at ~1.8V instead of being at 0V and 3.3V.

I've made a test using STM32-Arduino and deploying the following sketch on my board:

```cpp
void setup() {
  Serial.begin(115200);
  pinMode(A2, OUTPUT);
}

void loop() {
  digitalWrite(A2, LOW);
  delay(5000);
  digitalWrite(A2, HIGH);
  delay(5000);
}
```

Here I could successfully measure the same pin alternating between 0V and 3.3V.

Is there something missing in order to get this working correctly or is it simply a bug?

 - OS: Gentoo Linux
 - Toolchain: Zephyr SDK, latest release

### PR DESCRIPTION
**Pull request by erwango**
To work efficiently, SPI_NSS pins require pull-up configuration.
Fix this for whole STM32 series.

Fixes #17998

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

**Review by chrta — APPROVED** [review]

**Comment by avisconti** [comment]
@erwango 
LGTM.
BTW, can you summarize here what tests you performed and on which target?

**Comment by erwango** [comment]
@avisconti, tested on F4 and L4, in master mode, with various CPOL, CPHA configurations.
